### PR TITLE
Add song genre field

### PR DESCRIPTION
**Pull request by razzeee**
Based on @sorianog work.

I've added a table with genres, a list of those can be fetched via it's own api endpoint.
Endpoints responding with a song should now also respond with a genreId and a text genre.

Frontend has been updated to ask for an genre on upload, which has to be set. Will default to the first item at the moment.

Genres will also be shown on each song item in the frontend.

I did not test what happens with songs that have no genre set, so that might cause problems at the moment.

There is also no way to change a genre on an already uploaded song.

### CODE REVIEW

File: database/seeds/DatabaseSeeder.php [hunk 1]
```diff
@@ -11,6 +11,6 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
-        // $this->call(UsersTableSeeder::class);
```
Comment (sorianog) [inline database/seeds/DatabaseSeeder.php]: Make sure to clean up commented out code if no longer needed

File: database/seeds/GenresTableSeeder.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        ['name' => 'Easy listening'],
+        ['name' => 'Electronic'],
+        ['name' => 'Folk'],
+        ['name' => 'Hip hop'],
+        ['name' => 'Jazz'],
+        ['name' => 'Latin'],
+        ['name' => 'Metal'],
+        ['name' => 'Pop'],
+        ['name' => 'R&B and soul'],
+        ['name' => 'Rock']
```
Comment (sorianog) [inline database/seeds/GenresTableSeeder.php]: Pretty good list of Genres. What did you end up using to determine the ones to use?

File: database/seeds/DatabaseSeeder.php (hunk 1 above)
Comment (Razzeee) [reply database/seeds/DatabaseSeeder.php]: Your commenting on removed/changed code. This side of things is how it was before my changes.

File: database/seeds/GenresTableSeeder.php (hunk 2 above)
Comment (Razzeee) [reply database/seeds/GenresTableSeeder.php]: I think I used the ones from wikipedia and cleaned them up a bit. But I'm not 100% sure, as this part was done some weeks ago.

File: database/seeds/DatabaseSeeder.php (hunk 1 above)
Comment (sorianog) [reply database/seeds/DatabaseSeeder.php]: Whoops I was on my phone while doing this review...lol

**Comment by Razzeee** [comment]
Fixed the layout and added editing for already created songs.

We do need to figure out how to migrate the old songs. Give them a genre "None" by default and don't let anyone save new Songs with that genre.

**Comment by Razzeee** [comment]
Rebased and added another commit that handles the migration from not having any genres on songs to having to add them.
We initially set the values in our database to null, but enforce setting genre via all the frontends.

I've tested this locally migrating a database with a song from `development` to this version. I was able to successfully do that and add a genre on that song.

As far as I know this should be everything that's needed.

**Comment by Razzeee** [comment]
Conflict resolved.

**Comment by Razzeee** [comment]
Should I close this?

**Pull request closed without merging**